basedosdados_api/api/v1/models.py

```diff
@@ -28,20 +28,67 @@ class Meta:
 
 class Coverage(models.Model):
     id = models.UUIDField(primary_key=True, default=uuid4)
-    area = models.ForeignKey("Area", on_delete=models.CASCADE, related_name="coverages")
-    temporal_coverages = models.ManyToManyField(
-        "TemporalCoverage", related_name="coverages"
+    table = models.ForeignKey(
+        "Table",
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="coverages",
     )
-
-    def __str__(self):
-        return str(self.temporal_coverages)
+    raw_data_source = models.ForeignKey(
+        "RawDataSource",
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="coverages",
+    )
+    information_request = models.ForeignKey(
+        "InformationRequest",
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="coverages",
+    )
+    column = models.ForeignKey(
+        "Column",
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="coverages",
+    )
+    key = models.ForeignKey(
+        "Key", blank=True, null=True, on_delete=models.CASCADE, related_name="coverages"
+    )
+    area = models.ForeignKey("Area", on_delete=models.CASCADE, related_name="coverages")
 
     class Meta:
         db_table = "coverage"
         verbose_name = "Coverage"
         verbose_name_plural = "Coverages"
         ordering = ["id"]
 
+    def __str__(self):
+        return str(self.id)
+
+    def clean(self) -> None:
+        # Assert that only one of "table", "raw_data_source", "information_request", "column" or
+        # "key" is set
+        count = 0
+        if self.table:
+            count += 1
+        if self.raw_data_source:
+            count += 1
+        if self.information_request:
+            count += 1
+        if self.column:
+            count += 1
+        if self.key:
+            count += 1
+        if count != 1:
+            raise ValidationError(
+                "One and only one of 'table', 'raw_data_source', 'information_request', 'column' or 'key' must be set."  # noqa
+            )
```

Comment:
@rdahis só confirma pra mim pfvr se é essa mesmo a validação. se for, tá pronto

Reply:
Não tenho certeza se entendo as implicações aqui. Mas parece certo.

Reply:
Não o pode ter o mesmo coverage identificado via `id` né? Porque se quiser preencher um coverage igual não teria problema né? Exemplo: duas tabelas com `coverage=[sa.br, 2000-2010]` (em pseudo-notação aí)

Reply:
isso, exatamente! não pode ter o exato mesmo registro do Coverage (mas podem ter registros com atributos iguais, só ids diferentes)

Reply:
Então manda bala aí!